subprojects/core/src/test/groovy/org/gradle/api/internal/tasks/options/OptionReaderTest.groovy

```diff
@@ -631,6 +708,22 @@ class OptionReaderTest extends Specification {
         }
     }
 
+    public static class TestClassWithListPropertyField {
+        @Option(option = 'customOptionName', description = "custom description")
+        final ListProperty<String> field1
```

Comment:
🤔 How does this code compile? It's a `final` field that is never initialized... Even if it's not a problem for this test, it would be good to make sure that the code is well formed. Maybe use an `abstract` getter instead to declare the properties?

Reply:
To address this and the other comment on the naming (`field1`, `field2`), I added the test cases for `ListProperty` and `SetProperty` to the existing test  "handles property field options". Given that this is a unit test, the option declaration might be tolerable? 

The numbered naming helps with the expected order in which the task options are put out (alphabetical order), so I think leaving it as is makes understanding the test easier.

Reply:
I see. If numbering is about enforcing an order required for the test to work, I think it would be good to make that explicit either by a comment or making this more obvious in the names, like `_01listProperty` and `_02setProperty` or something like this.

Reply:
Ah, I see, you've adopted an existing naming scheme... I think we could improve on this, but since you've now merged the new cases into the existing test, I think this is fine.